docs/HowToHashiCorpVault.md

```diff
@@ -0,0 +1,270 @@
+```
+
+### Create a NuoDB Secrets Engine
+
+The final step is to create a [Secrets KV Engine](https://www.vaultproject.io/docs/secrets/kv/index.html) to store the NuoDB keys.
+All keys will be base64 encoded.
+
+```
+kubectl exec -it -n vault vault-0 -- sh
+$ vault secrets enable -version=2 -path=nuodb.com kv
+```
+
+## Adding NuoDB Keys to HashiCorp Vault
+
+In the previous step [Generating NuoDB Keys](##generating-nuodb-keys) we generated a set of keys required to run NuoDB.
+We saved those keys in `/tmp/keys`.
+In this step, we will save these keys in HashiCorp Vault.
+
+```
+kubectl exec -it -n vault vault-0 -- \
+    vault kv put nuodb.com/TLS \
+    tlsClientPEM=`cat /tmp/keys/nuocmd.pem | base64` \
+    tlsCACert=`cat /tmp/keys/ca.cert | base64` \
+    tlsKeyStorePassword=changeIt \
+    tlsTrustStorePassword=changeIt \
+    tlsKeyStore=`cat /tmp/keys/nuoadmin.p12 | base64` \
+    tlsTrustStore=`cat /tmp/keys/nuoadmin-truststore.p12 | base64`
+```
+
+## Configuration of NuoDB
+
+HashiCorp Vault uses annotations to identify pods that require a Vault Agent.
+For more information please consult the [Template Documentation](https://www.vaultproject.io/docs/platform/k8s/injector/annotations#vault-hashicorp-com-agent-inject-template)
+
+### Annotations for the NuoDB Admin Tier
+
+The NuoDB admin tier requires all four key files. 
+For convenience, we list them here again:
+- `nuodb-keystore.p12` containing the X509 that identifies the admin;
+- `nuodb-truststore.p12` usually containing the root CA and the primordial admin user;
+- `ca.cert` containing the public certificate of the Certificate Authority;
+- `nuocmd.pem` which is the private/public keypair for the primordial admin user.
+
+Both the keystore and the truststore PKCS12 files are password protected.
+The passwords need to be exported as environmental variables.
+
+```
+$ cat vault-annotations-admin.yaml
+admin:
+  podAnnotations:
+    vault.hashicorp.com/agent-inject: "true"
+    vault.hashicorp.com/agent-inject-secret-ca.cert: nuodb.com/TLS
+    vault.hashicorp.com/agent-inject-secret-nuoadmin-truststore.p12: nuodb.com/TLS
+    vault.hashicorp.com/agent-inject-secret-nuoadmin-truststore.password: nuodb.com/TLS
+    vault.hashicorp.com/agent-inject-secret-nuoadmin.p12: nuodb.com/TLS
+    vault.hashicorp.com/agent-inject-secret-nuoadmin.password: nuodb.com/TLS
+    vault.hashicorp.com/agent-inject-secret-nuocmd.pem: nuodb.com/TLS
+    vault.hashicorp.com/agent-inject-template-ca.cert: |
+      {{- with secret "nuodb.com/TLS" -}}
+        {{ .Data.data.tlsCACert | base64Decode }}
+      {{- end }}
+    vault.hashicorp.com/agent-inject-template-nuoadmin-truststore.p12: |
+      {{- with secret "nuodb.com/TLS" -}}
+        {{ .Data.data.tlsTrustStore | base64Decode }}
+      {{- end }}
+    vault.hashicorp.com/agent-inject-template-nuoadmin-truststore.password: |
+      {{- with secret "nuodb.com/TLS" -}}
+        export NUODB_TRUSTSTORE_PASSWORD=”{{ .Data.data.tlsTrustStorePassword }}”
```

Comment:
This can't be correct. I remember that this was needed because previously we didn't have the file content expansion syntax in `nuoadmin.conf` file. Can you try with a non-default password to make sure that this works properly? I.e `--set admin.options.truststore-password='$(</etc/nuodb/keys/nuoadmin-truststore.p12)'`